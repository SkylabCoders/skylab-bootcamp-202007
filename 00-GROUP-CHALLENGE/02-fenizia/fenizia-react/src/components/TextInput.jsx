import React from "react";

function TextInput(props) {
  return (
    <label htmlFor={props.name}>
      <input
        className="input__search"
        name={props.name}
        type={props.type ? props.type : "text"}
        placeholder={props.placeholder}
        value={props.value}
        onChange={props.onChange}
      />
    </label>
  );
}

export default TextInput;
