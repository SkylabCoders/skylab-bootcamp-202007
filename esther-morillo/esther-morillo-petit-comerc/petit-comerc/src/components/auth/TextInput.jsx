import React from 'react';
import PropTypes from 'prop-types';

function TextInput (props) {
    let wrapperClass = "form-group";
    if(props.error.length > 0) {
        wrapperClass += " has error";
    }

    return (
        <div className={wrapperClass}>
            <label htmlFor={props.id}>{props.label}</label>
            <div className="field">
                <input
                    id={props.id}
                    type="text"
                    name={props.name}
                    className="form-control"
                    value={props.value}
                    onChange={props.onChange}
                />
            </div>
            {props.error && <div className="">{props.error}</div>}
        </div>
    );
}

TextInput.propTypes = {
    id: PropTypes.string,
    name: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    value: PropTypes.string,
    error: PropTypes.string
};

TextInput.defaultProps = {
    error: ''
}

export default TextInput;