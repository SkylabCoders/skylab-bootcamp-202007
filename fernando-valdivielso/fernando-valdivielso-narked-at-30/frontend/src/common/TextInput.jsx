import React from 'react';
import '../styles/Textinput.css';

function TextInput(props) {
	return (
		<label htmlFor={props.name} className='label'>
			{props.label}:{' '}
			<input
				className='label__input'
				name={props.name}
				type={props.type}
				placeholder={props.placeholder}
				value={props.value}
				onChange={props.onChange}
			/>
		</label>
	);
}

export default TextInput;
