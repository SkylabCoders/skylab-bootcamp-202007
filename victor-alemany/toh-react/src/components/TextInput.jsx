import React from 'react';

function TextInput(props) {
	return (
		<label htmlFor={props.name}>
			{'Name: '}
			<input
				name={props.name}
				type="text"
				placeholder={props.placeholder}
				value={props.value}
				onChange={props.onChange}
			/>
		</label>
	);
}

export default TextInput;