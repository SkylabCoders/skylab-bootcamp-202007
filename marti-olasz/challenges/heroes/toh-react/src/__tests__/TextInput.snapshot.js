import renderer from 'react-test-renderer';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import TextInput from '../components/TextInput';

describe('TextInputs snapshot', () => {
	const tree = renderer.create(
		<Router>
			<TextInput />
		</Router>
	);
	it('Should match', () => {
		expect(tree.toJSON()).toMatchSnapshot();
	});
});
