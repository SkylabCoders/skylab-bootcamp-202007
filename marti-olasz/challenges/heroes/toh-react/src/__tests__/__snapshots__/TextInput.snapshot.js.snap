// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputs snapshot Should match 1`] = `
<label>
  Name: 
  <input
    type="text"
  />
</label>
`;
